packages/collector/test/tracing/messaging/node-rdkafka/test.js

```diff
@@ -0,0 +1,290 @@
+/*
+ * (c) Copyright IBM Corp. 2022
+ */
+
+'use strict';
+
+/**
+ * Important notes
+ * ---------------
+ *
+ * * The Producer as a stream can only have span correlation if the Writtable option objectMode is set to true.
+ * Otherwise, there is no way to append the Instana headers to it.
+ * * The Producer, as stream or standard API cannot propagate span correlation when headerFormat is set to 'binary'.
+ * More info here: https://github.com/Blizzard/node-rdkafka/pull/935.
+ * * If the option dr_cb is not set to true, we cannot guarantee that a message was sent, but a span with a successful
+ * sent message will be created.
+ */
+
+const path = require('path');
+const { expect } = require('chai');
+const { fail } = expect;
+const {
+  tracing: { constants }
+} = require('@instana/core');
+
+const {
+  tracing: { supportedVersion }
+} = require('@instana/core');
+
+const config = require('../../../../../core/test/config');
+const { expectExactlyOneMatching, retry, delay, stringifyItems } = require('../../../../../core/test/test_util');
+const ProcessControls = require('../../../test_util/ProcessControls');
+const globalAgent = require('../../../globalAgent');
+const { verifyHttpRootEntry, verifyHttpExit } = require('@instana/core/test/test_util/common_verifications');
+
+let mochaSuiteFn;
+
+const producerEnableDeliveryCbOptions = ['true', 'false'];
+const producerApiMethods = ['standard', 'stream'];
+const consumerApiMethods = ['standard', 'stream'];
+
+const { getCircularList } = require('@instana/core/test/test_util/circular_list');
+const getNextProducerMethod = getCircularList(producerApiMethods);
+const getNextConsumerMethod = getCircularList(consumerApiMethods);
+const getNextDeliveryCb = getCircularList(producerEnableDeliveryCbOptions);
+
+if (!supportedVersion(process.versions.node)) {
+  mochaSuiteFn = describe.skip;
+} else {
+  mochaSuiteFn = describe;
+}
+const retryTime = config.getTestTimeout() * 2;
+
+const topic = 'rdkafka-topic';
+
+mochaSuiteFn('tracing/messaging/node-rdkafka', function () {
+  this.timeout(config.getTestTimeout() * 4);
+
+  globalAgent.setUpCleanUpHooks();
+  const agentControls = globalAgent.instance;
+
+  describe('tracing enabled, no suppression', function () {
+    const producerControls = new ProcessControls({
+      appPath: path.join(__dirname, 'producer'),
+      port: 3216,
+      useGlobalAgent: true,
+      env: {
+        RDKAFKA_PRODUCER_DELIVERY_CB: getNextDeliveryCb()
```

Comment:
We are currently testing for _"tracing enabled, no suppression"_ only with `RDKAFKA_PRODUCER_DELIVERY_CB: true`. I was wondering if we also want to test with `false`. My initial expectation here was that we have to iterate over the `producerEnableDeliveryCbOptions`.

Reply:
Good catch. Will do it.